web_poet/page_inputs/response.py

```diff
@@ -0,0 +1,30 @@
+from typing import Union
+
+import attrs
+
+from web_poet.mixins import SelectableMixin, UrlShortcutsMixin
+from web_poet.page_inputs.browser import BrowserResponse
+from web_poet.page_inputs.http import HttpResponse
+from web_poet.page_inputs.url import ResponseUrl
+
+
+@attrs.define
+class AnyResponse(SelectableMixin, UrlShortcutsMixin):
+    """A container that holds either :class:`~.BrowserResponse` or :class:`~.HttpResponse.`"""
+
+    response: Union[BrowserResponse, HttpResponse]
+
+    @property
+    def url(self) -> ResponseUrl:
+        """URL of the response."""
+        return self.response.url
```

Comment:
Should we convert self.response.url to str here or in heuristics? Because this is failed https://github.com/zytedata/zyte-spider-templates/pull/28/files#diff-ca9971971b7e9f63c22373022f4e32c974e800a6bc36f9e7e490453ed1038951R53

Reply:
Good catch! This should be `ResponseUrl`, so it's the same type when accessing `BrowserResponse.url` and `HttpResponse.url`. I'll fix the spider template code for this instead.